lib/compat/wordpress-6.0/class-wp-webfonts.php

```diff
@@ -51,12 +80,288 @@ public function init() {
 			$this->stylesheet_handle = 'webfonts';
 			$hook                    = 'wp_enqueue_scripts';
 		}
+
+		add_action( 'init', array( $this, 'register_current_template_filter' ) );
+		add_action( 'init', array( $this, 'register_globally_used_webfonts' ) );
+
+		add_action( 'switch_theme', array( $this, 'invalidate_all_used_webfonts_cache' ) );
+		add_action( 'save_post_wp_template', array( $this, 'invalidate_used_webfonts_cache' ) );
+		add_action( 'save_post_wp_template_part', array( $this, 'invalidate_used_webfonts_cache' ) );
+
+		add_filter( 'the_content', array( $this, 'register_webfonts_used_in_content' ) );
+
+		add_filter( 'rest_request_after_callbacks', array( $this, 'maybe_invalidate_globally_used_webfonts_cache' ), 10, 3 );
+
 		add_action( $hook, array( $this, 'generate_and_enqueue_styles' ) );
 
 		// Enqueue webfonts in the block editor.
 		add_action( 'admin_init', array( $this, 'generate_and_enqueue_editor_styles' ) );
 	}
 
+	/**
+	 * Invalidates cache on global styles update.
+	 *
+	 * @param WP_REST_Response $response The response class.
+	 * @param WP_REST_Server   $handler The request handler.
+	 * @param WP_REST_Request  $request The request class.
+	 *
+	 * @return WP_REST_Response
+	 */
+	public function maybe_invalidate_globally_used_webfonts_cache( $response, $handler, $request ) {
+		if ( 'GET' === $request->get_method() || ! is_a( $handler['callback'][0], 'WP_REST_Global_Styles_Controller' ) ) {
+			// We only want to intercept requests that update global styles!
+			return $response;
+		}
+
+		$this->invalidate_globally_used_webfonts_cache();
```

Comment:
This is unnecessary. Solely updating the option as I'm doing in the next line is enough.